lib/src/lifecycle_module.dart

```diff
@@ -127,10 +138,84 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
     }.forEach(_logLifecycleEvents);
   }
 
+  /// If this module is in a transition state, this is the Span capturing the
+  /// transition state.
+  Span get activeSpan => _activeSpan;
+
+  /// Set internally by this module for the load span so it can be used as a
+  /// `Reference` to other spans after the span is finished.
+  SpanContext _loadContext;
+
+  /// Set internally by the parent module if this module is called by [loadChildModule]
+  SpanContext _childOfContext;
+
+  /// Builds a span that conditionally applies a followsFrom reference if this module
+  /// was loaded by a parent module.
+  ///
+  /// Returns `null` if no globalTracer is configured.
+  Span _startTransitionSpan(String operationName) {
+    final tracer = globalTracer();
+    if (tracer == null) {
+      return null;
+    }
+
+    List<Reference> references = [];
+    if (_childOfContext != null) {
+      references.add(new Reference.followsFrom(_childOfContext));
+    }
+
+    return tracer.startSpan(
+      operationName,
+      references: references,
+      tags: _defaultTags,
+    );
+  }
+
+  /// Creates a span from the start of [load] until now.
+  ///
+  /// This is different from [onLoad] as this indicates that whichever asynchronous
+  /// processes required to make this module interactible (such as any data needed
+  /// to populate its store) will be finished by this point, whereas it might
+  /// only be started in [onLoad].
+  ///
+  /// Any [tags] specified will be added to the span which results from this call.
+  @protected
+  void specifyStartupTiming(
+    StartupTimingSpecifier specifier, {
+    Map<String, dynamic> tags: const {},
```

Comment:
Can we allow passing in other `references` so this reference can be associated with whatever triggered it, if that thing is traced?

Reply:
There's already `followsFrom` reference to the load span.